tests/training/trainer_test.py

```diff
@@ -127,6 +127,26 @@ def setup_method(self):
         self.validation_data_loader.index_with(self.vocab)
 
 
+class ZeroGradientsBackwardCallback(TrainerCallback):
+    """
+    Zeros all gradients after backpropagation.
+    """
+
+    def on_backward(
+        self,
+        trainer: "GradientDescentTrainer",
+        loss: torch.FloatTensor,
+        backward_called: bool,
+        **kwargs,
+    ) -> bool:
+        if not backward_called:
+            loss.backward()
+            for param in trainer.model.parameters():
+                param.grad *= 0.0
```

Comment:
Is that really the best way to do that?
```suggestion
                param.zero_()
```
I don't know for sure, but I would guess that `zero_()` is faster.